frontend/src/concepts/trustyai/useManageTrustyAICR.ts

```diff
@@ -1,64 +1,78 @@
 import React from 'react';
-import useTrustyAINamespaceCR, {
-  isTrustyAIAvailable,
-  taiHasServerTimedOut,
-} from '~/concepts/trustyai/useTrustyAINamespaceCR';
-import { createTrustyAICR, deleteTrustyAICR } from '~/api';
+import useTrustyAINamespaceCR from '~/concepts/trustyai/useTrustyAINamespaceCR';
+import {
+  assembleSecret,
+  createSecret,
+  createTrustyAICR,
+  deleteSecret,
+  deleteTrustyAICR,
+} from '~/api';
+import { getTrustyStatusState } from '~/concepts/trustyai/utils';
+import { TRUSTYAI_SECRET_NAME } from '~/concepts/trustyai/const';
+import useTrustyBrowserStorage from '~/concepts/trustyai/content/useTrustyBrowserStorage';
+import { TrustyDBData, TrustyStatusStates } from './types';
+
+export type UseManageTrustyAICRReturnType = {
+  statusState: TrustyStatusStates;
+  installCRForNewDB: (secretData: TrustyDBData) => Promise<void>;
+  installCRForExistingDB: (secretName: string) => Promise<void>;
+  deleteCR: () => Promise<void>;
+};
 
 const useManageTrustyAICR = (namespace: string): UseManageTrustyAICRReturnType => {
   const state = useTrustyAINamespaceCR(namespace);
-  const [cr, loaded, serviceError, refresh] = state;
-
-  const [installReqError, setInstallReqError] = React.useState<Error | undefined>();
+  const [cr, , , refresh] = state;
+  const successDetails = useTrustyBrowserStorage(cr);
 
-  const isAvailable = isTrustyAIAvailable(state);
-  const isProgressing = loaded && !!cr && !isAvailable;
-  const error = installReqError || serviceError;
+  const statusState = getTrustyStatusState(state, successDetails);
 
-  const [disableTimeout, setDisableTimeout] = React.useState(false);
-  const serverTimedOut = !disableTimeout && taiHasServerTimedOut(state, isAvailable);
-  const ignoreTimedOut = React.useCallback(() => {
-    setDisableTimeout(true);
-  }, []);
+  const installCRForExistingDB = React.useCallback<
+    UseManageTrustyAICRReturnType['installCRForExistingDB']
+  >(
+    async (secretName) => {
+      await createTrustyAICR(namespace, secretName).then(refresh);
+    },
+    [namespace, refresh],
+  );
+  const installCRForNewDB = React.useCallback<UseManageTrustyAICRReturnType['installCRForNewDB']>(
+    async (data) => {
+      const submitNewSecret = async (dryRun: boolean) => {
+        await Promise.all([
+          createSecret(assembleSecret(namespace, data, 'generic', TRUSTYAI_SECRET_NAME), {
+            dryRun,
+          }),
+          createTrustyAICR(namespace, TRUSTYAI_SECRET_NAME, { dryRun }),
+        ]);
+      };
 
-  const showSuccess = React.useRef(false);
-  if (isProgressing) {
-    showSuccess.current = true;
-  }
+      await submitNewSecret(true);
+      await submitNewSecret(false);
+      await refresh();
+    },
```

Comment:
should the `submitNewSecret(false)` only run if the dry run was successful? it looks like it will run regardless

Reply:
If it fails the dry run, an error is thrown from the `Promise.all` which breaks `submitNewSecret` which breaks `installCRForNewDB` which is caught by the caller of the whole thing.